### DISCUSSION TITLE
how to avoid online downing while i have got the offline wight files

### DISCUSSION THREAD

**Discussion by arctanbell** [towhee-io/towhee#2660] [Q&A]
I want to] use the llm.llama2, but downloading files named like
![image](https://github.com/towhee-io/towhee/assets/8346786/a7d2814c-ec4b-49e5-b889-2c49e4dcb155)
so I downloaded the files llama-2-7b-chat.ggmlv3.q4_0 and llama-2-7b-chat.ggmlv3.q4_1 and placed them at the same dir, however the download still continued
![image](https://github.com/towhee-io/towhee/assets/8346786/80f93522-4e59-42df-acd4-a88aeaa1a462)

**Comment by junjiejiangjjj** [comment]
You can also use the local path of a model file, which can be ran by llama-cpp-python.
```python
chat = ops.LLM.Llama_2({YOUR_LOCAL_FILE_PATH}, n_ctx=4096, max_tokens=200)
```

**Reply by arctanbell** [reply]
i have try like this,
> p = (
>  pipe.input('question', 'docs', 'history')
> .map(('question', 'docs', 'history'), 'prompt', ops.prompt.question_answer())
>  .map('prompt', 'answer', ops.LLM.Llama_2('./llama-2-7b-chat.ggmlv3.q4_0.bin',n_ctx=4096, max_tokens=200))
>  .output('answer')
)
get the erro like:

> gguf_init_from_file: invalid magic number 67676a74
> error loading model: llama_model_loader: failed to load model from ./llama-2-7b-chat.ggmlv3.q4_0.bin
> llama_load_model_from_file: failed to load model
i search this erro,get this answer [https://huggingface.co/TheBloke/Llama-2-13B-chat-GGML/discussions/14](url), but have to change the version of llama.cpp,how to solver it ?

**Comment by junjiejiangjjj** [comment]
Install the previous version before this one: https://github.com/ggerganov/llama.cpp/commit/dadbed99e65252d79f81101a392d0d6497b86caa
